### PR TITLE
Encapsulate relative position logic in UITextInput.on_event

### PR DESCRIPTION
**Pull request by pushfoo**
* Add relative position helpers
* Use position helpers to clean up UITextInput.on_event
* Some light refactoring for readability
* Improve related docstrings and comments

### CODE REVIEW

**Comment by pushfoo** [comment]
@DigiDuncan Let me know if the relative position helper seems to belong in `Rect` or if I somehow missed one like it already existing.

File: arcade/gui/widgets/__init__.py [hunk 1]
```diff
@@ -167,6 +178,20 @@ def on_update(self, dt):
...
+            mouse_event: Any :py:class:`UIMouseEvent`.
+        Returns:
+            ``None`` if outside the widget or coords relative to
+             the widget's bottom left.
+        """
+        pos = mouse_event.pos
+        if not self.rect.point_in_rect(pos):
+            return None
+        return Vec2(pos[0] - self.left, pos[1] - self.bottom)
+
```
Comment (DragonMoffon) [inline arcade/gui/widgets/__init__.py:194]: Update to use the new `Rect` method.

**Review by DragonMoffon — CHANGES_REQUESTED** [review]

File: arcade/gui/widgets/__init__.py (hunk 1 above)
Comment (pushfoo) [reply arcade/gui/widgets/__init__.py:194]: I didn't update it yet because I want to give it more thought after rereading the UI code and considering the following:
1. 3.9 features will be our new minimum
2. The new rect method seems to perform more ops
3. It may add even more call overhead

**Comment by eruvanos** [comment]
Hi, I am unsure about the state of the changes, due to the conflicts and the lag of an open issue, I would maybe close this PR?
@pushfoo let me know, if you thing the UITextInput requires the refactoring, from my point of view, right now it is kind of explicit. But I am always open for a second opinion.

**Comment by pushfoo** [comment]
> I would close this PR
TL;DR: Agreed.

This thing was an exploratory kludge. I want to review your recent UI event proposals for controllers before attempting this + thoroughly review `Rect` since:
1. I need to in general
2. Dragon's "use rect" suggestion seemed to consume more operations at first glance
3. The fact I said "seemed" means we need to bench it
